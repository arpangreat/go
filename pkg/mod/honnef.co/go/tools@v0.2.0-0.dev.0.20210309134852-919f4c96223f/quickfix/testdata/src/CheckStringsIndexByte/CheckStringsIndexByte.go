package pkg

import (
	"bytes"
	"strings"
)

func fn() {
	strings.Index("", "0")            // want `could use strings.IndexByte instead of strings.Index`
	strings.LastIndex("", "0")        // want `could use strings.LastIndexByte instead of strings.LastIndex`
	strings.IndexByte("", '0')        // want `could use strings.Index instead of strings.IndexByte`
	strings.LastIndexByte("", '0')    // want `could use strings.LastIndex instead of strings.LastIndexByte`
	bytes.Index(nil, []byte{'0'})     // want `could use bytes.IndexByte instead of bytes.Index`
	bytes.LastIndex(nil, []byte{'0'}) // want `could use bytes.LastIndexByte instead of bytes.LastIndex`
	bytes.Index(nil, []byte("0"))     // want `could use bytes.IndexByte instead of bytes.Index`

	strings.Index("", "µ")
	strings.Index("", "00")
	strings.LastIndex("", "00")
	bytes.LastIndex(nil, []byte{'0', '0'})
	bytes.Index(nil, []byte("µ"))
}
