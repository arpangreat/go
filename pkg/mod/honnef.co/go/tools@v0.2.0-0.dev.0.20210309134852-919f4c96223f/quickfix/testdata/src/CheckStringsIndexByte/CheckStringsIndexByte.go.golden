package pkg

import (
	"bytes"
	"strings"
)

func fn() {
	strings.IndexByte("", '0')     // want `could use strings.IndexByte instead of strings.Index`
	strings.LastIndexByte("", '0') // want `could use strings.LastIndexByte instead of strings.LastIndex`
	strings.Index("", "0")         // want `could use strings.Index instead of strings.IndexByte`
	strings.LastIndex("", "0")     // want `could use strings.LastIndex instead of strings.LastIndexByte`
	bytes.IndexByte(nil, '0')      // want `could use bytes.IndexByte instead of bytes.Index`
	bytes.LastIndexByte(nil, '0')  // want `could use bytes.LastIndexByte instead of bytes.LastIndex`
	bytes.IndexByte(nil, '0')      // want `could use bytes.IndexByte instead of bytes.Index`

	strings.Index("", "µ")
	strings.Index("", "00")
	strings.LastIndex("", "00")
	bytes.LastIndex(nil, []byte{'0', '0'})
	bytes.Index(nil, []byte("µ"))
}
